### DISCUSSION TITLE
Can you uninstall Asus Hotplug Controller

### DISCUSSION THREAD

**Discussion by XynoCrest** [seerge/g-helper#1592] [General]
Just a quick question as I can't seem to find any documentation about it anywhere - Does Ghelper need Asus Hotplug Controller to be installed or can we safely uninstall it?

**Comment by seerge** [comment]
@XynoCrest g-helper doesn't need it :) rest - is up to you

**Reply by XynoCrest** [reply]
Then I am uninstalling it, why have more crap around :) Thanks always for the quick and helpful reply!

**Comment by Kikimahe** [comment]
Hi, 

Note that if _ASUS Hotplug controller_ is disabled from Task Scheduler or uninstalled, Windows displays that the dGPU is an ejectable device.

If Hotplug Controller has been uninstalled or if you are doing a fresh installation of Windows, you need to install _the Asus Touch pad driver_ to fix it. This driver installs Asus Hot Plug controller.

![Sans titre-1](https://github.com/seerge/g-helper/assets/142881210/32245fae-c41b-4ce7-8e80-67f23b07cdbf)

**Reply by XynoCrest** [reply]
Thank you so much for this! This weird Eject GPU bug has been haunting me forever (ever since I disabled it in autoruns), and even after hours of searching online I couldn’t find a possible solution so I just learned to live with it. Thank you again for this gem of an info. Now I don’t have to worry about accidentally Ejecting my GPU out of my laptop :)

**Reply by Kikimahe** [reply]
You are welcome. :) 

Several months ago, following several new installations of Windows, I understood that this issue (GPU eject) could be solved by installing Asus Touchpad Drivers. It was weird but it fixed it and I didn't understand why ... 

Yesterday I understood that Asus Hotplug Controller was the cause of the issue.
And it can **only** be installed with Asus Touchpad driver.

To answer your first question "Can you uninstall Asus Hotplug Controller ?" I would say definitely not. ;-)

**Reply by seerge** [reply]
@Kikimahe that's an interesting find, that Touchpad driver installs it :)

**Reply by seerge** [reply]
@Kikimahe for me Hotplug Controller doesn't hide anything related to ejecting GPU btw

![Screenshot 2024-02-04 154828](https://github.com/seerge/g-helper/assets/5920850/c14587a9-2429-4a51-846b-b2d915db1c23)

**Reply by Kikimahe** [reply]
Oh, that's very weird. ,'( 
Did you restart Windows after installing Asus Touch pad?

This works for me and others.
Maybe this only fixes for Nvidia cards.

**Reply by seerge** [reply]
@Kikimahe i did :) furthermore, i reboot straight away in Standard mode (so GPU is already there). It doesn't stop windows from thinking it's ejectable.

**Reply by Kikimahe** [reply]
I disabled the hotplug controller task and restart.
And the Nvidia card came back as ejectable.

Is there anything I can do to help you find what Hotplug controller is doing on my system?

![Sans titre-1](https://github.com/seerge/g-helper/assets/142881210/359a0801-8bb8-4ce8-afb8-eb6a8d333fe4)

**Reply by klofi** [reply]
Hello @Kikimahe, ASUS Zephyrus G16 2024 nVidia user here, I have uninstalled AC, disabled trigger of ASUS Hotplug Controller in Task Scheduler and rebooted. After reboot, I verified in Task Manager > Details, that the AsHotplugCtrl.exe is no longer running and I do not have the GPU as ejectable device. So perhaps it works differently on G16 2024, will report if I see any issues.

@seerge I want to thank you for this awesome tool. I already knew of G-Helper as it was actually the reason why I even considered buying an ASUS laptop, ASUS should pay you a commission 😄! (I know they won't, so I'll send at least support fee instead). I have been giving AC a try for a week before installing G-Helper on my new laptop and now cannot imagine going back to use AC. It's so much quicker and easier to use, and after fresh boot the RAM usage went from 7 GB to 4.7 GB... Also I really appreciate the extensive documentation and FAQ! One thing that I would recommend to update is better (un)install steps for new users, so all the info is in one place as I really didn't want to go through hassle of reinstalling clean OS (but looking at all the steps below, it might have been a quicker option 😃).

This is what I found out I had to do to get rid of all ASUS bloatware and have everything still working properly (so far) and on a brand new Zephyrus G16 2024:
* Covered by [Requirements > Recommendations (optional)](https://github.com/seerge/g-helper/wiki/Requirements#recommendations-optional)
  * Uninstall Armoury Crate 
  * Uninstall ASUS Smart Display Control 
  * Uninstall MyASUS
* Covered by [How do I stop the Armoury Crate install popup appearing every time I press the M4 / Rog key?](https://github.com/seerge/g-helper/wiki/FAQ#how-do-i-stop-the-armoury-crate-install-popup-appearing-every-time-i-press-the-m4--rog-key)
  * Disable services, so M4 key does not show annoying popup every time. I really think this should move from FAQ to [Requirements > Recommendations (optional)](https://github.com/seerge/g-helper/wiki/Requirements#recommendations-optional), as why would anyone uninstall AC and not do this step?
* Not covered
  * Uninstall ASUS Hotplug Controller (or disable in Task Scheduler)
  * Uninstall ASUS Monitor Control (or disable in Task Scheduler)
  * Uninstall GlideX Service Installer (in Settings > Apps) + delete folder in `C:\Program Files\ASUS\GlideX`
  * Uninstall GlideX (in Settings > Apps)
  * Delete task ASUS WDL Helper (Task Scheduler >  Task Scheduler Library > ASUS) + delete folder `C:\Program Files\ASUS\ASUS WDL Helper`
    * It seems to be related to Windows Dynamic Lightning, but even killing the .exe task runs does not have any impact on Settings > Personalization > Dynamic Lightning. Also noticed that even after using the ASUS official AC uninstall tool, there's still Armoury Crate listed as app in "Background light control" in the "Dynamic Lightning" Windows setting and could not find a way to get rid of that. No impact on G-Helper whatsoever.
  * Delete task Aura Wallpaper Service (Task Scheduler >  Task Scheduler Library > ASUS)
  * Delete task P508PowerAgent_sdk (Task Scheduler >  Task Scheduler Library > ASUS)
    * This one was really odd, task tried to run `C:\Program Files (x86)\ASUS\ArmouryDevice\dll\ShareFromArmouryIII\Mouse\ROG STRIX CARRY\P508PowerAgent.exe` (I never connected any ASUS mouse to laptop)

To disable trigger of a task to first verify functionality before deleting task: open task > Triggers > Edit > uncheck Enabled. 

There's still these 3 tasks in Task Scheduler I have rather left untouched, as they don't really seem to use resources and trigger .exe in `C:\Windows\System32`:
* ASUS Optimization
* ASUS Update Checker 2.0
* AsusSystemAnalysis

**Reply by Kikimahe** [reply]
Good job! :)

I have a feeling this ejectable GPU issue only affects the Tuf series.

You can disable these last 3 tasks. I haven't had any problems with them. Watch my screenshoot above. 
Deleting them is not useful, because they will be reinstalled with each installation of Asus drivers

Also download Autoruns64 and check if you still have ASUS processes or services : 
I remember that I still had a service or process dependent on Armory Crate that launched at startup. However, I have never installed AC on a clean installation of Windows...
I deleted it with Autoruns64.
